src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -349,6 +372,35 @@ HashSet<ITypeSymbol> typeStack
 			return MutabilityInspectionResult.NotMutable( unauditedReasonsBuilder.ToImmutable() );
 		}
 
+		private MutabilityInspectionResult InspectInterface(

```

Comment:
I'm a little bit confused.  I can see that this new code applies to type parameters on interfaces (it is called from the `TypeKind.Interface` switch block).  But I don't see how we end up examining the type parameters for inheritance / abstract classes.  Is it possible to get an error of `MutabilityCause.IsMutableTypeParameter` when inheriting from a base class rather than implementing an interface?